mailbox/noise.go

```diff
@@ -659,30 +685,26 @@ func (b *Machine) GenActTwo() ([ActTwoSize]byte, error) {
 	// At this point, we've authenticated the responder, and need to carry
 	// out the final step primarily to obtain their long-term public key
 	// and initialize the DH handshake.
-	//
-	// However, we also want to send the client data they may need for
-	// authentication (if present) encrypted with strong forward secrecy.
+
+	// We include the size of tha auth data that we will send along to the
```

Comment:
Weird protocol flow IMO: why do we need to send the size of a message that we'll send at a future time? We can just use the normal length prefix scheme to do that, packaging this payload within the auth payload we send anyway. 

We should also only do this if we're on the _newer_ handshake version.